app/indexer/LoreIndexer.ts

```diff
@@ -1,86 +1,51 @@
-// import { DesiegeResolver } from "../resolvers";
 import { Event } from "../entities/starknet/Event";
 import { Context } from "../context";
-import { Indexer } from "./../types";
 import { Contract } from "starknet";
 import { uint256ToBN } from "starknet/utils/uint256";
 import fetch from "node-fetch";
-
 import LoreABI from "../abis/Lore.json";
-import { toBN } from "starknet/utils/number";
+import BaseContractIndexer from "./BaseContractIndexer";
 
-export default class LoreIndexer implements Indexer<Event> {
-  private CONTRACTS = [
-    "0x521151a3e28d2efff7d7d34c4f8bbe45c4c9f37424bae4dc63836fd659c51e6"
-  ];
-  // @ts-ignore
-  private context: Context;
-  private contract: Contract = new Contract(LoreABI as any, this.CONTRACTS[0]);
+const CONTRACT = "0x06894a6766b4763d8bea8d43f433d25e577ed8bf057942c861df4e9951282c64";
 
+export default class LoreIndexer extends BaseContractIndexer {
+  private contract: Contract = new Contract(LoreABI as any, CONTRACT);
```

Comment:
Do we need to pass in Provider here?
e.g.
```
 this.contract = new Contract(LoreABI as any, CONTRACT, new Provider({
  network: NETWORK === "goerli" ? "goerli-alpha" : "mainnet-alpha"
}));

Reply:
Good idea! Just added.